streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -1070,6 +1071,11 @@ public void shouldTransitState() {
         assertEquals(ACTIVE_RESTORING, changelogReader.state());
         assertEquals(mkSet(tp, tp1, tp2), consumer.assignment());
         assertEquals(mkSet(tp1, tp2), consumer.paused());
+
```

Comment:
Wow, there is a LOT going on in this test...encroaching on 180 lines, yikes. Could you split this out into its own test to make it more clear what is being tested/that we have coverage for this edge case -- it's hard to get a grasp on whether we're actually testing everything when it's all crammed into a single test.

I know you didn't write this monolith to begin with, but we should strive always to improve on the sins of our ancestors 😉 More seriously though, covering multiple things in one test can be useful eg with integration tests where the setup & teardown alone take a significant chunk of time. But I would imagine/hope a unit test like this runs fairly quickly so it probably makes the most sense to optimize for comprehension

Sorry for the slightly pedantic speech, if it's any consolation I bet Bruno would have said more or less the same thing lol

Reply:
Done